### ISSUE TITLE
Doesn't handle incompatible value property

### ISSUE THREAD

**Issue by LouisJS** [archriss/react-native-render-html#154]
Everytime i'm facing some HTML using property such as 'initial', 'normal', ect.
It is isn't handled properly by this library

```
Warning: Failed prop type: Invalid prop `textDecorationColor` supplied to `Text`: initial
Warning: Failed prop type: Invalid prop `display` of value `inline` supplied to `Text`, expected one of ["none","flex"]
Warning: Failed prop type: Invalid prop `letterSpacing` of type `string` supplied to `Text`, expected `number`.
Bad object: {
  "letterSpacing": "normal"
}
Warning: Failed prop type: Invalid prop `textDecorationStyle` of value `initial` supplied to `Text`, expected one of ["solid","double","dotted","dashed"].
Failed prop type: Invalid prop `textAlign` of value `start` supplied to `Text`, expected one of ["auto","left","right","center","justify"].
```

I can't possibly ignore all these styling.

Using `react-native-render-html`: `3.9.3`,

**Comment by Exilz** [comment]
I think we should set up some kind of object that holds every conversion that needs to be done when a style property is available both in the browser and react native but with different accepted values.

The current logic where `'18px'` is converted into `18` just doesn't cut it. We need to remove specific code and handle all of this into a single helper.

This should be one of the last major steps towards being able to render _any_ HTML string without crashing.

**Comment by iamdurui** [comment]
I also have this crash

**Comment by sorakrisc** [comment]
any follow up on this?

**Comment by TheNuclearCat** [comment]
For me its an invalid propety in 'letterSpacing'

**Comment by MrLich** [comment]
any one fixed this bug successful . I have faced it with props display 'inline': should be one of (flex, none)

**Comment by zedtux** [comment]
I have this issue too :

```
Warning: Failed prop type: Invalid prop `letterSpacing` of type `string` supplied to `ForwardRef(Text)`, expected `number`.
```

**Comment by huangdou0204** [comment]
I have this issue too
Invalid property caused a crash

`Warning: Failed prop type: Invalid prop `textAlign` of value `start` supplied to `ForwardRef`, expected one of ["auto","left","right","center","justify"].`

**Comment by khaledwj90** [comment]
I fixed this issue by adding ignoredStyles={['letter-spacing']} Prop.
https://github.com/archriss/react-native-render-html#ignoring-html-content
`
 <HTML html={html_content} ignoredStyles={['letter-spacing']}/>
`

**Comment by zhugy-yami** [comment]
I have this issue too!
Please follow on this

**Comment by zhugy-yami** [comment]
I have this issue too!
Please follow up on this
v4.1.1

**Comment by ronyv** [comment]
Had the same issue with this library and spend almost 2 days trying to prevent the app from crashing. At the end i've fixed this issue in a different way. I have used the inbuilt WebView component for html rendering. Two issues i faced with WebView component are: 
1. Dynamic height for the view based on the html content.
2. to make the clickable links to open in an external browser.

I've fixed both the above issues in the following way.

`
const script = `
<script>
	window.location.hash = 1;
    var calculator = document.createElement("div");
    calculator.id = "height-calculator";
    while (document.body.firstChild) {
        calculator.appendChild(document.body.firstChild);
    }
	document.body.appendChild(calculator);
    document.title = calculator.scrollHeight;
</script>
`;
const style = `
<style>
body, html, #height-calculator {
    margin: 0;
    padding: 0;
}
#height-calculator {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
}
</style>
`;`

`<WebView
                 ref="webview"
                 originWhitelist={['*']}
                 source={{html: this.state.htmlcontent + style + script}}
                 style={{ height: this.state.contentHeight}}
                 scrollEnabled={false}
                 onNavigationStateChange={this.onNavigationChange.bind(this)}>
               </WebView>`


`onNavigationChange(event) {
   if(event.navigationType && event.navigationType === 'click') {
     this.refs.webview.stopLoading();
     Linking.openURL(event.url);
   }
   else if (event.title) {
     const htmlHeight = Number(event.title) //convert to number
     this.setState({contentHeight:htmlHeight});
   }
 }`

Hope this helps...

**Comment by ds8k** [comment]
What would be the simplest way of completely ignoring the `style` attribute entirely?

**Comment by saadzaman** [comment]
ds8k !! was wondering the same. Ignoring styles fixed this issue. I ignored this by using this prop allowedStyles={[]}

**Comment by ManigandanRaamanathan** [comment]
> ds8k !! was wondering the same. Ignoring styles fixed this issue. I ignored this by using this prop allowedStyles={[]}

This solution worked well.

**Comment by priyanka931** [comment]
Thanks @saadzaman !! Good to go

**Comment by yuezonglun** [comment]
ignoredStyles={['font-family', 'display','text-align',]}
allowedStyles={["none","flex"]}

**Comment by jsamr** [comment]
I have implemented a bunch of fixes, see #382, which should also fix what you've all encountered here. If you happen to still have an issue after release 4.2.2, please open an new ticket. Closing now, expecting release very soon.

**Comment by jsamr** [comment]
This issue will be fixed in version 4.2.2.
A pre-release is already available if you want this fix now, see #385.
